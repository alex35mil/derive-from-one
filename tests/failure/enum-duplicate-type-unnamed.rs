use derive_from_one::FromOne;

#[derive(FromOne)]
pub enum T {
    A(u32),
    B(u32),
}

fn main() {
    let _: T = T::from(1);
}
