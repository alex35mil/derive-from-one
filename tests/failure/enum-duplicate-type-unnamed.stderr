error[E0308]: mismatched types
  --> tests/failure/enum-duplicate-type-unnamed.rs:10:24
   |
10 |     let _: T = T::from(1);
   |                ------- ^ expected `T`, found integer
   |                |
   |                arguments to this function are incorrect
   |
note: associated function defined here
  --> $RUST/core/src/convert/mod.rs
   |
   |     fn from(value: T) -> Self;
   |        ^^^^
help: try wrapping the expression in a variant of `T`
    |
10  |     let _: T = T::from(T::A(1));
    |                        +++++ +
10  |     let _: T = T::from(T::B(1));
    |                        +++++ +
